pkg/scheduler/core/scheduling_queue.go

```diff
@@ -183,11 +191,51 @@ func NewPriorityQueue() *PriorityQueue {
 	pq := &PriorityQueue{
 		activeQ:        newHeap(cache.MetaNamespaceKeyFunc, util.HigherPriorityPod),
 		unschedulableQ: newUnschedulablePodsMap(),
+		nominatedPods:  map[string][]*v1.Pod{},
 	}
 	pq.cond.L = &pq.lock
 	return pq
 }
 
+// addNominatedPodIfNeeded adds a pod to nominatedPods if it has a NominatedNodeName and it does not
+// already exist in the map. Adding an existing pod is not going to update the pod.
+func (p *PriorityQueue) addNominatedPodIfNeeded(pod *v1.Pod) {
+	nnn := NominatedNodeName(pod)
+	if len(nnn) > 0 {
+		for _, p := range p.nominatedPods[nnn] {
+			if p.Name == pod.Name && p.Namespace == pod.Namespace {
```

Comment:
The change looks good overall.

Not sure if there's any change that name + namespace is not sufficient? Then we may need also respect UID here.

Reply:
That's a good point. I think checking UID should be enough, i.e., we don't need to check all the three fields, name and namespace and UID.

Reply:
Yep, +1 for UID only

Reply:
Done. PTAL. Thanks.